File: program/program/find-the-maximum-sum-of-a-contiguous-subarray-(kadane's-algorithm)/find_the_maximum_sum_of_a_contiguous_subarray_(kadane's_algorithm).php
<?php
function kadaneAlgorithm($array, $size) {
  $max = 0;
  $current = 0;
  for($i=0; $i<$size; $i++) {
    $current = $current + $array[$i];
    
    if ($current < 0)
      $current = 0; 
    
    if($max < $current)
      $max = $current; 
  }
  return $max;
}

//test
$array = array(1, 2, 3, -2, 5);
$size = sizeof($array);
$result = kadaneAlgorithm($array, $size);
echo "Maximum SubArray is: ".$result."\n";
?>

File: program/program/find-the-maximum-sum-of-a-contiguous-subarray-(kadanes-algorithm)/FindTheMaximumSumOfAContiguousSubarray(kadane'sAlgorithm).swift
class KadaneAlgorithm {
    func maxSubarraySum(_ nums: [Int]) -> Int {
        var maxEndingHere = nums[0]
        var maxSoFar = nums[0]
        
        for i in 1..<nums.count {
            maxEndingHere = max(nums[i], maxEndingHere + nums[i])
            maxSoFar = max(maxSoFar, maxEndingHere)
        }
        
        return maxSoFar
    }
}

let kadane = KadaneAlgorithm()
let input = [1, 2, 3, -2, 5]
let result = kadane.maxSubarraySum(input)
print("Maximum sum of a contiguous subarray: \(result)")
